server/functions/basebuilding/fn_resupplybuilding.sqf

```diff
@@ -95,6 +95,23 @@ if (_resupplyType == "MULTISANDBAG" && _supplyType == "BuildingSupplies") then {
 //No supplies, no point continuing execution.
 if (_supplies == 0) exitWith {};
 
+private _currentSupplies = _supplySource getVariable ["para_g_current_supplies", 0];
+private _supplyConsumptionRate = _supplySource getVariable ["para_g_supply_consumption_rate", 1];
+
+private _proposedSuppliesSum = _currentSupplies + _supplies;
+private _proposedRemainingTimeSeconds = _proposedSuppliesSum / _supplyConsumptionRate;
+private _maxTimeLimit = 7200; // max time limit a building can have, 2 hours in seconds
+
+// Check if the current lifetime would exceed the time limit
+if ( _proposedRemainingTimeSeconds >= _maxTimeLimit) then {
+	// if the supplies that are being added will exceed 2 hours for the building, trim the supplies so that it only reaches 2 hours max. This will still consume the item!!!
```

Comment:
split comment over two lines please cos it's quite long, we like to try and keep line length below 100 characters for readability and maintenance purposes

Example PR comment